### DISCUSSION TITLE
Different poses in faceswap?

### DISCUSSION THREAD

**Discussion by rotilho** [lllyasviel/Fooocus#1142] [Q&A]
I'm playing with faceswap however I can't make the character look in a different direction. Are there any way to achieve that?

**Comment by aryamanmajumdar** [comment]
I've been having trouble with this too. Tried playing around a lot with prompts, negative prompts, the weight parameter, etc. And a bunch of other things.

A generated front-facing face that I want to use for faceswapping always results in the face in the output image being front-facing (i.e. having the same pose as the 'input face'). 

Looked into ReActor for SD. Would it be possible to integrate or implement that?

Edit: Found a solution that will work for some cases. Put the 'body' image through PyraCanny, and the 'face' image through FaceSwap, and include a prompt. 

PyraCanny gets the edge map and FaceSwap gets the face. And the prompt generates an image based on the PyraCanny-extracted edge map and applies the face you extract using FaceSwap. (Badly explained, but I think you get it)

**Comment by tbuyle** [comment]
I did some successfull tests using multiple images :

Try generating a front and side view of the same face an then used them both a input images at the same time.
Varying the weight of one or the other change the angle.

![image](https://github.com/lllyasviel/Fooocus/assets/467350/db8704cb-e18a-48e0-b5f6-8aeae1e9f0c9)

**Reply by rotilho** [reply]
That makes sense but how did you generate the same photo from multiple angles?

**Reply by tbuyle** [reply]
I did it in one single generation - I don't remember the exact prompt.

**Reply by ghost** [reply]
@tbuyle But how did you manage to generate a side view of the same face, looking the same? I'm trying to do this at the moment. Or did you use ImagePrompt to generate a new face with a central and side view, instead of using face swap?

**Reply by tbuyle** [reply]
1. Create a first front portrait as reference. (Use Extreme Speed  to itterate quickly and find a face you like)
2. Use the below template to generate the portait in 4 positions, using 2 Image prompts : the portrait you juste made with FaceSwap and the below template with CPDS. Adapt the text prompt to somthing with "multiple portaits in different angles"
3. Split the resulting image in 4 new images.
4. Upscale them with 1.5x
5. Use 2 or 3 of them with Faceswap, adjusting Weight to change the angle.

![face-template-4angles](https://github.com/lllyasviel/Fooocus/assets/467350/6ed4f436-d7b0-48c9-bdf0-53a495579c0d)

**Reply by ghost** [reply]
Thanks for the help, Thomas!

**Reply by juliux32** [reply]
@tbuyle It kinda works and at the same time, it doesn't... Yes, it generates faces in these 4 positions 99% of the time without any extras in the background, BUT it does not generate a person's face used in FaceSwap... Don't know maybe I am doing something wrong but here's my BEST attempt...
![3814112](https://github.com/lllyasviel/Fooocus/assets/21074960/84465c4d-fc83-44a5-b0c9-a025ab731bfb)
![2024-01-08_11-34-45_2221](https://github.com/lllyasviel/Fooocus/assets/21074960/16ce163f-9972-4f6b-9001-3c80ec9d4f0a)

**Reply by tbuyle** [reply]
> @tbuyle It kinda works and at the same time, it doesn't... Yes, it generates faces in these 4 positions 99% of the time without any extras in the background, BUT it does not generate a person's face used in FaceSwap... Don't know maybe I am doing something wrong but here's my BEST attempt... 

Sorry for the late answer...

The tricky part is indeed to get the 4 angle shot from the first picture.

From the look of it, you'd probably want to reduce a bit the weight and stop at of the template. And perhaps give more weight to the face.

But of she is a real person, it would possibly be easier to get 4 pictures from different angle directly rather than generate them. My workflow is more adapted to a fully virtual character, using a generated face as first picture (and using part of the original prompt to generate the 4 angles in addition to the image prompt)

**Comment by marc2608** [comment]
https://www.youtube.com/watch?v=82bkNE8BFJA

**Comment by Alexmswe** [comment]
I found some tools that have a[ look-like feature](https://www.deepswap.ai/ai-girl-generator), which generates different poses based on prompts and then swaps faces. This is the result of my attempt.
![face swap image](https://github.com/lllyasviel/Fooocus/assets/167491808/6647e4b9-27a8-44df-951e-e8d49a12c8a2)
![faceswap image](https://github.com/lllyasviel/Fooocus/assets/167491808/26f7ac5d-8bfa-4b84-90b3-64eb0fb3094b)

**Comment by kalle07** [comment]
in a1111 i use as extension "reactor" it is similar to "faceswaper" (not the same like in fooocus).
you can use one face or calculate it from maybe 5, it works realy good with small faces.

https://civitai.com/articles/5002/face-swap-with-adetailer-and-reactor
